### ISSUE TITLE
Cannot change project icon on expo.dev

### ISSUE THREAD

**Issue by olarcher** [expo/eas-cli#2135]
### Build/Submit details page URL

https://expo.dev/accounts/oliver.larcher/projects

### Summary

I cannot find any option for changing the icon for a particular project on expo.dev. In the project settings menu, there is an option to change the display name under "Project display info", but not the icon. It appears that the icon is initially set when uploading a build with icon for the first time, but it is not updated then the app icon changes.

### Managed or bare?

Managed

### Environment

```
  expo-env-info 1.0.5 environment info:
    System:
      OS: macOS 14.1.1
      Shell: 5.9 - /bin/zsh
    Binaries:
      Node: 20.9.0 - /opt/homebrew/bin/node
      Yarn: 1.22.19 - /opt/homebrew/bin/yarn
      npm: 10.2.3 - /opt/homebrew/bin/npm
    SDKs:
      iOS SDK:
        Platforms: DriverKit 23.0, iOS 17.0, macOS 14.0, tvOS 17.0, watchOS 10.0
    IDEs:
      Android Studio: 2022.2 AI-222.4459.24.2221.9862592
      Xcode: 15.0.1/15A507 - /usr/bin/xcodebuild
    npmPackages:
      expo: ^49.0.0 => 49.0.6 
      react: 18.2.0 => 18.2.0 
      react-dom: 18.2.0 => 18.2.0 
      react-native: 0.72.3 => 0.72.3 
      react-native-web: ~0.19.6 => 0.19.7 
    npmGlobalPackages:
      eas-cli: 5.9.1
    Expo Workflow: managed
```

```
✔ Check Expo config for common issues
✔ Check package.json for common issues
✔ Check dependencies for packages that should not be installed directly
✔ Check for common project setup issues
✔ Check npm/ yarn versions
✔ Check Expo config (app.json/ app.config.js) schema
✔ Check for legacy global CLI installed locally
✔ Check that native modules do not use incompatible support packages
✔ Check that native modules use compatible support package versions for installed Expo SDK
✖ Check that packages match versions required by installed Expo SDK

Detailed check results:

Some dependencies are incompatible with the installed expo version:
  expo-dev-client@2.4.6 - expected version: ~2.4.12
  expo-updates@0.18.11 - expected version: ~0.18.17
  react-native@0.72.3 - expected version: 0.72.6
Your project may not work correctly until you install the correct versions of the packages.
Fix with: npx expo install --fix
Found outdated dependencies
```

### Reproducible demo or steps to reproduce from a blank project

Go to `https://expo.dev/accounts/<ACCOUNT>/projects` and try to change a project's icon.

**Comment by tylerhammer** [comment]
Any chance you figured this out? It's been driving me nuts for some time.

**Comment by olarcher** [comment]
No, unfortunately. I'm starting to wonder if changing the icon used to be an option that was later removed, as the icons I still see in my project list are the ones for projects I created over two years ago. It seems new projects don't get the app logo defined at first build time by default anymore.

**Comment by DavidAmyot** [comment]
+1 I can't find anywhere how to change the actual project icon on expo.dev. The actual app's icon is fine, but I just want to change the project icon on expo.dev and I can't find out how?

**Comment by chaadow** [comment]
+1

**Comment by bjarheim** [comment]
+1

**Comment by joaomellogomes** [comment]
The same with me

**Comment by lletfrix** [comment]
+1

**Comment by eduardocesb** [comment]
+1

**Comment by stesvis** [comment]
Nothing yet?

**Comment by oliviercperrier** [comment]
+1

**Comment by mirague** [comment]
+1

**Comment by michaek** [comment]
@olarcher, you marked this as needs review last November. Based on the history here, this is a real problem with expo.dev that needs to be resolved, not just reviewed. It's very frustrating to see outdated and missing icons in the management UI despite multiple builds/deploys with updated app icons, and yet have no way to resolve it.

**Comment by szdziedzic** [comment]
Hi,

Thanks for highlighting this issue. We are aware that we are missing this capacity on our website, we are thinking of adding it in the future, but we currently don't have the engineering capacity to do that. Generally, we understand that it might be problematic and want to solve it, but I can't promise any ETA at the moment.

**Comment by michaek** [comment]
Thanks for the update - just knowing that it's not in indefinite limbo is helpful!

**Comment by olarcher** [comment]
Still relevant.

**Comment by Spharian** [comment]
+1

**Comment by ancabanca123** [comment]
+1 
The icon for my App changed in stores. I would like to be able to change it in expo too.

**Comment by jacklin213** [comment]
Not stale, still awaiting fix

**Comment by brugaspar** [comment]
Any news about this?

**Comment by rafatrace** [comment]
Just adding one more voice to the problem. We've been updating our projects and 2 out of 3 projects have new app icons that have nothing to do with actual.

**Comment by ArtanBajqinca** [comment]
+1

**Comment by nikita1469** [comment]
I had the same issue but when i ran expo-doctor it show this warning
Detailed check results from expo-doctor: This project contains native project folders but also has native configuration properties in app.config.js, indicating it is configured to use Prebuild. When the android/ios folders are present, EAS Build will not sync the following properties: orientation, icon, userInterfaceStyle, splash, ios, android, plugins. 

Advice: Add '/android' and '/ios' to your .gitignore file if you intend to use CNG / Prebuild. Learn more: https://docs.expo.dev/workflow/prebuild/#usage-with-eas-build 

So i just removed /ios and /android folders from my project and it works

**Comment by olarcher** [comment]
@nikita1469 this issue is about changing the project icon of your projects listed on the expo.dev, not the app icon.

**Comment by brugaspar** [comment]
+1

**Comment by michaek** [comment]
Turn off the bot for this. It's obviously needed and should be scheduled.

**Comment by rafatrace** [comment]
Let us upload a icon, or even simpler, a text field for an URL with the icon, pleaseeee 😃

**Comment by michaek** [comment]
@szdziedzic I'm arbitrarily picking on you for this one again, but just because you didn't throw someone else under the bus! 

I want to be really clear that the inability to change the app icon on expo.dev is a regression. It used to be that the configured app icon would show up on expo.dev when new builds were uploaded. When that behavior ended, it left the icon from the last build in place, so many projects may now have outdated (thus misleading) icons. This is just broken behavior, and it makes expo.dev harder to use.

No new functionality is necessary here - just restoring the old behavior would be sufficient.

**Comment by tomimarkus991** [comment]
+1

**Comment by alfredogazperi** [comment]
+1

**Comment by arthurolga** [comment]
+1

**Comment by LinFeiLong** [comment]
+1 too

**Comment by michaek** [comment]
This is an essential task to restore a feature that was removed inadvertently. Turn off the bot so we don't have to bury the actual issue under messages to keep it "active". Also, get this fixed. It is becoming ridiculous that you have broken this and simply ignored it for so long.

**Comment by michaek** [comment]
I'm glad this exists so we can fix icons that have been wrong for years, but I'm curious why you removed the previous behavior of setting the project icon during builds.

**Comment by kadikraman** [comment]
I'm sorry this went unsupported for so long. The original project icon upload relied on classic updates, which are no longer available, and unfortunately there wasn’t a straightforward migration path due to technical limitations. The new image upload is a ground-up implementation, and now works across projects, orgs, and user profiles.

We’re a small team with many priorities to juggle, and in hindsight, this definitely should’ve gotten more attention sooner. I completely understand the frustration, and I appreciate everyone’s patience while we worked through it.

**Comment by michaek** [comment]
Thanks for the explanation! However it works is ok with me, and I'm really glad this is something we can adjust now.
